grakn-graph/src/main/java/ai/grakn/graph/internal/ResourceImpl.java

```diff
@@ -99,59 +94,11 @@ public Thing owner() {
      * @param value The value to store on the resource
      * @return The Resource itself
      */
-    private Resource<D> setValue(D value) {
-        try {
-            checkConformsToRegexes(value);
-
-            ResourceTypeImpl<D> resourceType = (ResourceTypeImpl<D>) type();
-            Schema.VertexProperty property = dataType().getVertexProperty();
-            //noinspection unchecked
-            vertex().propertyImmutable(property, castValue(value), vertex().property(property), (v) -> resourceType.getDataType().getPersistenceValue((D) v));
-            vertex().propertyUnique(Schema.VertexProperty.INDEX, generateResourceIndex(type().getLabel(), value.toString()));
-
-            return getThis();
-        } catch (ClassCastException e) {
-            throw GraphOperationException.invalidResourceValue(value, dataType());
-        }
-    }
-
-    /**
-     * Checks if all the regex's of the types of this resource conforms to the value provided.
-     *
-     * @throws GraphOperationException when the value does not conform to the regex of its types
-     * @param value The value to check the regexes against.
-     */
-    private void checkConformsToRegexes(D value){
-        //Not checking the datatype because the regex will always be null for non strings.
-        for (ResourceType rt : ((ResourceTypeImpl<D>) type()).superSet()) {
-            String regex = rt.getRegex();
-            if (regex != null && !Pattern.matches(regex, (String) value)) {
-                throw GraphOperationException.regexFailure(this, (String) value, regex);
-            }
-        }
-    }
-
-    /**
-     * This is to handle casting longs and doubles when the type allows for the data type to be a number
-     * @param value The value of the resource
-     * @return The value casted to the correct type
-     */
-    private Object castValue(Object value){
-        ResourceType.DataType<D> parentDataType = dataType();
-        if(parentDataType.equals(ResourceType.DataType.DOUBLE)){
-            return ((Number) value).doubleValue();
-        } else if(parentDataType.equals(ResourceType.DataType.LONG)){
-            if(value instanceof Double){
-                throw new ClassCastException();
-            }
-            return ((Number) value).longValue();
-        } else {
-            try {
-                return Class.forName(parentDataType.getName()).cast(value);
-            } catch (ClassNotFoundException e) {
-                throw new RuntimeException(ErrorMessage.INVALID_RESOURCE_CAST.getMessage(value, parentDataType.getName()));
-            }
-        }
+    private Resource<D> setValue(Object value) {
```

Comment:
Because `setValue` now takes in the "persisted value" i.e the value in the format in which it's going to be saved, and this can anything really.